feat: 0.76 Release Blog Post (#4279)
facebook/react-native-website: website/blog/2024-10-23-release-0.76-new-architecture.md

- [Removed the dependency on the react-native-community/cli](/blog/2024/10/23/release-0.76-new-architecture#removed-the-dependency-on-the-react-native-communitycli)
- [Android Apps are ~3.8Mb smaller thanks to Native Library merging](/blog/2024/10/23/release-0.76-new-architecture#android-apps-are-38mb-smaller-thanks-to-native-library-merging)
- [Updates to Minimum iOS and Android SDK requirements](/blog/2024/10/23/release-0.76-new-architecture#updates-to-minimum-ios-and-android-sdk-requirements)

<!--truncate-->

## Highlights

### React Native New Architecture by default

Starting from React Native 0.76, the New Architecture is enabled by default in your projects. The New Architecture is a rewrite of the internals of React Native to enable app developers to develop high quality native applications using React.

Today, we’re excited to declare the New Architecture ready for production use.

This change is a milestone in the evolution of React Native, and we invite you to read the dedicated blogpost to understand what's included in the New Architecture and how it will shape the future of React Native: [The New Architecture is Here](/blog/2024/10/23/the-new-architecture-is-here).

> 

Reviewer: Fixed

- [Android Apps are ~3.8Mb smaller thanks to Native Library merging](/blog/2024/10/23/release-0.76-new-architecture#android-apps-are-38mb-smaller-thanks-to-native-library-merging)
- [Updates to Minimum iOS and Android SDK requirements](/blog/2024/10/23/release-0.76-new-architecture#updates-to-minimum-ios-and-android-sdk-requirements)

<!--truncate-->

## Highlights

### React Native New Architecture by default

Starting from React Native 0.76, the New Architecture is enabled by default in your projects. The New Architecture is a rewrite of the internals of React Native to enable app developers to develop high quality native applications using React.

Today, we’re excited to declare the New Architecture ready for production use.

This change is a milestone in the evolution of React Native, and we invite you to read the dedicated blogpost to understand what's included in the New Architecture and how it will shape the future of React Native: [The New Architecture is Here](/blog/2024/10/23/the-new-architecture-is-here).

### React Native DevTools

![React Native DevTools front-end](../static/blog/assets/0.76-react-native-devtools.jpg)

> We're releasing the first stable version of React Native DevTools, a reliable and familiar debugging tool. Key features include:
> 
> • **Familiar, web-aligned tooling**: A fully featured debugger based on Chrome DevTools with reliable breakpoints, watch values, step-through debugging, stack inspection, and a rich JavaScript console.
> • **Improved and integrated React DevTools**: The built-in React Components Inspector and Profiler work seamlessly, with faster and more reliable component highlighting.
> • **Improved UX**: A new Paused in Debugger overlay, summary warnings in LogBox, and fixed reconnection behavior.
> • **Instant launch**: React Native DevTools is ready by default with zero config, and can be opened from the in-app Dev Menu or via <kbd>j</kbd> to debug in the CLI server.
> 
> This new debugging experience is fundamentally different from previous options and [has changed](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=announcement%20talk%20%F0%9F%8E%AC.-,Compatibility,-React%20Native%20DevTools) compatibility. It's built on a new backend debugging stack, rebuilt from the ground up, and will support future feature additions like Performance and Network panels.
> 
> #### Phasing out logs in Metro
> 
> In our next release, we're removing forwarded logs in Metro to align with modern browser tooling, and remove old debugging integrations. Instead, use React Native DevTools' fully featured Console panel for logging. See more under our [FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=point%20is%20deprecated.-,FAQs,-React%20Native%20DevTools).

Reviewer 2: > [!Note]
> _**Damn, we shipped 😅**_ — I will follow up.

Manual pass to fix lost context (keeping target word length). Also, replace unicode bullet characters with Markdown list items.

I'd rather reduce the total bullets we have here (saving for dedicated post) than compact the meaning away as much.

```suggestion
We're releasing the first stable version of React Native DevTools, our new default debugging experience. Key features include:

- **Familiar, web-aligned tooling**: A fully featured debugger based on Chrome DevTools with reliable breakpoints, watch values, step-through debugging, stack inspection, and a rich JavaScript console. We've fixed .
- **Improved and integrated React DevTools**: The built-in React Components Inspector and Profiler work seamlessly, with faster and more reliable component highlighting.
- **Improved UX**: A new Paused in Debugger overlay, summary warnings in LogBox, and fixed reconnection behavior.
• **Instant launch**: React Native DevTools is ready by default with zero config, and can be opened from the in-app Dev Menu or via <kbd>j</kbd> to debug in the CLI server.

React Native DevTools is fundamentally different from previous debugging options, and [has changed](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=announcement%20talk%20%F0%9F%8E%AC.-,Compatibility,-React%20Native%20DevTools) compatibility. It's built on a new backend debugging stack, rebuilt from the ground up, and will support future feature additions like Performance and Network panels.

#### Phasing out logs in Metro

In our next release, we're removing forwarded logs in Metro to align with modern browser tooling, and remove old debugging integrations. Instead, use React Native DevTools' fully featured Console panel for logging. See more under our [FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=point%20is%20deprecated.-,FAQs,-React%20Native%20DevTools).
```

---
title: 'React Native 0.76 - New Architecture by default, React Native DevTools, Box Shadows and more'
authors: [blakef, cipolleschi, frankcalise, gabrieldonadel]
tags: [engineering]

> date: 2024-10-23

Reviewer 3: Is this the planned release date?
Just asking to figure out if we should plan a headline in [This Week In React](https://thisweekinreact.com/newsletter) (send on Wednesdays, usually end of day)

Reviewer 4: yes, that's the tentative release date, if everything goes according to our plans! :D

### Highlights

- [React Native New Architecture by default](#react-native-new-architecture-by-default)
- [React Native DevTools](#react-native-devtools)
- [Box Shadow and Filter style props](#box-shadow-and-filter-style-props)

### Breaking Changes

- [Removed the dependency on the react-native-community/cli](#removed-the-dependency-on-the-react-native-communitycli)
- [[Android] React Native C++ code is distributed as libreactnative.so](#android-react-native-is-distributed-as-a-single-library)
- [Updates to Minimum iOS and Android SDK requirements](#updates-to-minimum-ios-and-android-sdk-requirements)

<!--truncate-->

## Highlights

### React Native New Architecture by default

The New Architecture is finally here and starting from 0.76 is enabled by default in your projects.

Thanks to our collaboration with the whole community, we feel now confident to make this step. We verified that more than [880 libraries](https://reactnative.directory/?newArchitecture=true) are compatible with the New Architecture, including all the Expo Modules.

> To make sure your experience with the New Architecture will be great, we worked with [BlueSky](https://github.com/bluesky-social/social-app/releases/tag/1.92.0-na-rc.2) and Kraken who are already using it in production.

Reviewer 5: > To make sure your experience with the New Architecture will be great,

Cut this and combine this sentence with the previous one

> Thanks to the community's validation and testing, we now feel confident making this change. We verified that more than 880 libraries are compatible with the New Architecture, including all the Expo Modules. We also worked with companies like BlueSky and Kraken to validate the app migration, and they are already using it in production.

Also "all the Expo Modules". Is that any module written with the Expo Module API? Or the set of modules provided by Expo? If the latter, perhaps make it "all the Expo Modules in the Expo SDK"

> This change will be completely transparent to you: by upgrading to version 0.76, following the instruction in the [upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.0), you’ll also be turning on the New Architecture. We don’t expect any changes in behaviors in your app.

Author: I've reworked to sentence.

If for any reason you can’t move to the New Architecture, you can still opt-out from it.

On Android, you can modify the `gradle.properties` file and turn the `newArchEnabled` flag to `false`.

```diff
-newArchEnabled=false
+newArchEnabled=false
```

On iOS, you can opt-out from the New Architecture by installing Cocoapods dependencies with the command:

```bash
RCT_NEW_ARCH_ENABLED=0 bundle exec pod install
```

> As part of the work to make the New Architecture the default, we also revamped most sections of the [website](https://reactnative.dev) to reflect these changes. If you need the documentation of the [Legacy Native Modules](https://reactnative.dev/docs/0.76/legacy/native-modules-intro) and [Legacy Native Components](https://reactnative.dev/docs/0.76/legacy/native-components-android), you can find it in the Legacy Architecture section. Otherwise, the rest of the website assumes that you are looking for New Architecture documentation.

Reviewer 6: To avoid forcing internal links to open in new tab let's use relative links in blog post body:

```suggestion
As part of the work to make the New Architecture the default, we also revamped most sections of the website to reflect these changes. If you need the documentation of the [Legacy Native Modules](/docs/0.76/legacy/native-modules-intro) and [Legacy Native Components](/docs/0.76/legacy/native-components-android), you can find it in the Legacy Architecture section. Otherwise, the rest of the website assumes that you are looking for New Architecture documentation.
```

Linking back to the homepage feels a bit redundant, that's why I have also suggested skipping that anchor.

### React Native DevTools

![React Native DevTools front-end](../static/blog/assets/0.76-react-native-devtools.jpg)

We are shipping the first stable release of React Native DevTools, our new default debugging experience.

> We want the tooling you use to debug React across all platforms to be reliable, familiar, simple, and cohesive. React Native DevTools is an out-of-the-box experience that delivers on these principles — familiar browser-aligned developer tools that are deeply integrated with React Native.

Reviewer 2: This refers to the "reliable, familiar, simple, and cohesive" principles stated in the preceding sentence. "Key features" is separate.

Phasing out logs in Metro
In our next release, we’re also removing forwarded logs in Metro. We are doing this to align with modern browser tooling and to remove old, non-CDP, debugging integrations. Streamed application logs are not a pattern in web bundlers or in native build tools, and these are neither well-formatted nor searchable. Instead, please use React Native DevTools and its fully featured Console panel. See more under our [FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=point%20is%20deprecated.-,FAQs,-React%20Native%20DevTools).

> **Links**

Reviewer 4: ```suggestion
To learn more about the new React Native DevTools, here some useful resources:
```

Reviewer 2: nit: I'd rather keep this terse. As replied on the GDoc, I don't think this strongly establishes any new convention. Happy with another short heading, e.g. "Resources".

> - [Updated debugging docs](https://reactnative.dev/docs/debugging)

Reviewer 6: ```suggestion
- [Updated debugging docs](/docs/debugging)
```

- [Announcement talk at React Universe Conf](https://www.youtube.com/watch?v=OwivVpg6Luc)
- [Feedback thread and FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819)

### Box Shadow and Filter style props

> We’ve added two New-Architecture-only style props in 0.76 - `boxShadow` and `filter`. Both of these props exist on the web, and the team tried to stick to the specs when possible so that these props are predictable, familiar, and ultimately easier to adopt. As a result you can look up the web documentation to fully understand how these work, but there are a few important differences which are noted below.

Reviewer 4: ```suggestion
We’ve added two New Architecture-only style props in 0.76 - `boxShadow` and `filter`. Both of these props exist on the web, and the team tried to stick to the specs when possible so that these props are predictable, familiar, and ultimately easier to adopt. As a result, you can look up the web documentation to fully understand how these work, but there are a few important differences which are noted below.
```

#### `boxShadow`

`boxShadow` adds a shadow to an element, with the ability to control the position, color, size, and blurriness of the shadow. Check out the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow) for a full overview of each of these arguments, along with a fiddle to try them out for yourself. Below are a few examples of the shadows you can make.

![New boxShadow style prop](../static/blog/assets/0.76-boxshadow-example.png)

`boxShadow` can take either a string, which mimics the CSS syntax, or JS objects which can embed variables. For example the string `‘5 5 5 0 rgba(255, 0, 0, 0.5)’` and the object `{offsetX: 5, offsetY: 5, blurRadius: 5, spreadDistance: 0, color: ‘rgba(255, 0, 0, 0.5)’}` would yield the same box shadow.

> The difference between `boxShadow` and the [existing shadow functionality](https://reactnative.dev/docs/shadow-props) is existing shadows:

Reviewer 4: ```suggestion
There are some differences between the new `boxShadow` and the [existing shadow functionality](https://reactnative.dev/docs/shadow-props). The previous functionality:
```

> - Do not work on Android

Reviewer 4: ```suggestion
- Does not work on Android
```

- Cannot be [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#inset)

> - Do not have [spread](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#length)

Reviewer 4: ```suggestion
- Does not have [spread](https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#length)
```

> - Do not work on `View`s if there is no background color

Reviewer 4: ```suggestion
- Does not work on `View`s if there is no background color
```

> - Are separate props, so they cannot share web syntax

Reviewer 4: ```suggestion
- Is a separate prop, so it can't share the same web syntax
```

> **Limitations & Spec Deviations**

Reviewer 4: ```suggestion
#### Limitations & Spec Deviations
```

- The default shadow color is black, not the parent’s color
- Android normal shadows are supported on **Android 9+**
- Android inset shadows are supported on **Android 10+**

#### `filter`

`filter` adds certain graphical filters to an element. There are a mix of color filters that let you modify things like brightness, saturation, and hue as well as non-color filters that let you add blurs and shadows. Check out the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/filter) for a full overview of each individual filter function, along with a fiddle to try them out for yourself. Below is a hot dog image with various filters applied to it.

> ![From left to right: no filter, saturate filter, blur filter, invert filter](../static/blog/assets/0.76-filter-example.png)
> 
> <p align="center">From left to right: no filter, `saturate` filter, `blur` filter, `invert` filter</p>

Reviewer 6: Formatting nit, we can use `figure` and `figcaption` to improve the display a bit:

```suggestion
<figure>
  <img alt="Filters demonstration" src="/blog/assets/0.76-filter-example.png" />
  <figcaption>From left to right: no filter, `saturate` filter, `blur` filter, `invert` filter</figcaption>
</figure>
```

![Screenshot 2024-10-15 at 15 09 07](https://github.com/user-attachments/assets/d0ad52d4-9717-4536-a3a0-3dd020584336)

Like `boxShadow`, `filter` can take either a string, which mimics the CSS syntax, or an array of JS objects which can embed variables. For example the string `‘saturate(0.5) grayscale(0.25)’` and the array `[{saturate: 0.5}, {grayscale: 0.25}]` would yield the same filter.
`filter` has a `dropShadow` value which varies slightly from `boxShadow`. The biggest difference is that `dropShadow` is an alpha mask - so the shadow will only be cast by a pixel if it has a nonzero alpha component. `boxShadow`, on the other hand, will cast around the border box of the element, even if nothing is inside of it. Additionally, dropShadow does not have a spread distance parameter and cannot be inset (shadow cast inside of the element).

> **Limitations and Spec Deviations**

Reviewer 4: ```suggestion
#### Limitations and Spec Deviations
```

Decoupling React Native from the CLI allows us to move faster through releasing these projects independently.

If you are relying on the CLI in your daily workflow, make sure to explicitly [add the dependency](https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.0#RnDiffApp-package.json) on the CLI in your `package.json`:

```diff
//…
  “devDependencies”: {
   // …
   + “@react-native-community/cli”: “15.0.0”,
   + "@react-native-community/cli-platform-android": “15.0.0”,
   + "@react-native-community/cli-platform-ios": “15.0.0”,
  },
```

### [Android] React Native is distributed as a single library

React Native 0.76 will ship with a reduced number of native dynamic libraries, as we merged a lot of our native code into a single library called `libreactnative.so`.
This change comes with reduction in app size, and improvement in app startup performance on Android.

On the other hand, this is a breaking change for both app and library developers.

> App developers will have to update their Application’s onCreate as follows:

Reviewer 4: ```suggestion
App developers will have to update their Application’s `onCreate` as follows:
```

```diff
+import com.facebook.react.soloader.OpenSourceMergedSoMapping
import com.facebook.soloader.SoLoader

class MainApplication : Application(), ReactApplication {

  override fun onCreate() {
    super.onCreate()
+   SoLoader.init(this, OpenSourceMergedSoMapping)
-   SoLoader.init(this, false)
  }
}
```

This change is necessary in order to properly load `libreactnative.so` and is included in the upgrade-helper.

Library authors won’t be affected by this change, unless you have custom C++ code.
For a technical deep dive into this change, and suggestions for library authors, you can read more about it [in this dedicated post](https://github.com/react-native-community/discussions-and-proposals/discussions/816).

### Updates to Minimum iOS and Android SDK requirements

We have updated our minimum platform and SDK versions:

- iOS: [15.1](https://support.apple.com/en-gb/108051#151)
- Android: [SDK 24](https://developer.android.com/tools/releases/platforms#7.0) (Android 7)

> This change was announced earlier in the year when 0.75 [was released](https://reactnative.dev/blog/2024/08/12/release-0.75#last-version-supporting-minsdk-23-and-miniosversion-134). For more background, please see the dedicated posts for [Android](https://github.com/react-native-community/discussions-and-proposals/discussions/802) and [iOS](https://github.com/react-native-community/discussions-and-proposals/discussions/812).

Reviewer 6: ```suggestion
This change was announced earlier in the year when 0.75 [was released](/blog/2024/08/12/release-0.75#last-version-supporting-minsdk-23-and-miniosversion-134). For more background, please see the dedicated posts for [Android](https://github.com/react-native-community/discussions-and-proposals/discussions/802) and [iOS](https://github.com/react-native-community/discussions-and-proposals/discussions/812).
```

---
title: 'React Native 0.76 - New Architecture by default, React Native DevTools, Box Shadows and more'
authors: [blakef, cipolleschi, frankcalise, gabrieldonadel]
tags: [engineering]
date: 2024-10-23
---

Today we are excited to release React Native 0.76! This is a major milestone for React Native, with the New Architecture enabled by default, and the new React Native DevTools. This has been the culmination of 6 years of hard work from our incredible community of developers to make this day happen.

### Highlights

- [React Native New Architecture by default](#react-native-new-architecture-by-default)
- [React Native DevTools](#react-native-devtools)
- [Box Shadow and Filter style props](#box-shadow-and-filter-style-props)

### Breaking Changes

- [Removed the dependency on the react-native-community/cli](#removed-the-dependency-on-the-react-native-communitycli)
- [[Android] React Native C++ code is distributed as libreactnative.so](#android-react-native-is-distributed-as-a-single-library)
- [Updates to Minimum iOS and Android SDK requirements](#updates-to-minimum-ios-and-android-sdk-requirements)

<!--truncate-->

## Highlights

### React Native New Architecture by default

> The New Architecture is finally here and starting from 0.76 is enabled by default in your projects.

Reviewer 5: This sentence feels like it is missing some punctuation. 

Also, you don't need the "finally". It kinda disparages ourself. Just state confidently what **is**.

> Starting from 0.76, The New Architecture is enabled by default!

The New Architecture is finally here and starting from 0.76 is enabled by default in your projects.

> Thanks to our collaboration with the whole community, we feel now confident to make this step. We verified that more than [880 libraries](https://reactnative.directory/?newArchitecture=true) are compatible with the New Architecture, including all the Expo Modules.

Reviewer 5: > Thanks to our collaboration with the whole community

While we were part of the collaboration, this tone gives off some self-importance.

Just give the community the credit:

> Thanks to the community's validation and testing, we now feel confident making this change.

### Highlights

- [React Native New Architecture by default](/blog/2024/10/23/release-0.76-new-architecture#react-native-new-architecture-by-default)
- [React Native DevTools](/blog/2024/10/23/release-0.76-new-architecture#react-native-devtools)
- [Box Shadow and Filter style props](/blog/2024/10/23/release-0.76-new-architecture#box-shadow-and-filter-style-props)

### Breaking Changes

- [Removed the dependency on the react-native-community/cli](/blog/2024/10/23/release-0.76-new-architecture#removed-the-dependency-on-the-react-native-communitycli)
- [[Android] React Native C++ code is distributed as libreactnative.so](/blog/2024/10/23/release-0.76-new-architecture#android-react-native-is-distributed-as-a-single-library)
- [Updates to Minimum iOS and Android SDK requirements](/blog/2024/10/23/release-0.76-new-architecture#updates-to-minimum-ios-and-android-sdk-requirements)

<!--truncate-->

## Highlights

### React Native New Architecture by default

> The New Architecture is here. Starting from 0.76 the New Architecture is enabled by default in your projects.

Reviewer 5: ```suggestion
Starting from 0.76 the New Architecture is enabled by default in your projects.
```

Thanks to the community's validation and testing, we now feel confident making this change. We verified that more than [880 libraries](https://reactnative.directory/?newArchitecture=true) are compatible with the New Architecture, including all the Expo Modules in the Expo SDK. We also worked with companies like [BlueSky](https://github.com/bluesky-social/social-app/releases/tag/1.92.0-na-rc.2) and [Kraken](https://blog.kraken.com/product/engineering/kraken-new-architecture) to validate the app migration, and they are already using it in production.

This change will be completely transparent to you. Upgrade to version 0.76 by following the instructions in the [upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.0), and the New Architecture will be enabled by default. We don’t expect any changes in behaviors in your app.

If for any reason you can’t move to the New Architecture, you can still opt-out from it.

- On Android, you can modify the `gradle.properties` file and turn the `newArchEnabled` flag to `false`.

```diff

> -newArchEnabled=false

Reviewer 5: This should start off as `true`, right?

### React Native DevTools

![React Native DevTools front-end](../static/blog/assets/0.76-react-native-devtools.jpg)

We are shipping the first stable release of React Native DevTools, our new default debugging experience.
We want the tooling you use to debug React across all platforms to be reliable, familiar, simple, and cohesive. React Native DevTools is an out-of-the-box experience that delivers on these principles — familiar browser-aligned developer tools that are deeply integrated with React Native.
Key features:

- **Familiar, web-aligned tooling** — A fully featured debugger based on Chrome DevTools, providing reliable breakpoints, watch values, step through debugging, stack inspection, and a rich JavaScript console. These core features now work reliably and across reloads.

> - **An improved and integrated version of React DevTools** — The built-in React Components Inspector and Profiler are powerful tools to inspect and understand your React app. Both tools work the same as they do on the web. Component highlighting is faster and more reliable than before.

Reviewer 2: Can we keep the earlier wording for this? Helps maintain momentum in the sentence (at least, matches the writing style of the surrounding list items).

```suggestion
- **An improved and integrated version of React DevTools** — The built-in React Components Inspector and Profiler are powerful tools to inspect and understand your React app, and work just as they do on the web. Component highlighting is faster and more reliable than before.
```

> - **Improved UX** — See a new Paused in Debugger overlay, making it clear when your app is paused on a breakpoint. Warnings in LogBox are now displayed as a summary, and hidden when DevTools is attached.

Reviewer 2: ```suggestion
- **Improved UX** — See a new *Paused in Debugger* overlay, making it clear when your app is paused on a breakpoint. Warnings in LogBox are now displayed as a summary, and hidden when DevTools is attached.
```

#### Phasing out logs in Metro

In our next release, we’re also removing forwarded logs in Metro. We are doing this to align with modern browser tooling and to remove old, non-CDP, debugging integrations. Streamed application logs are not a pattern in web bundlers or in native build tools, and these are neither well-formatted nor searchable. Instead, please use React Native DevTools and its fully featured Console panel. See more under our [FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=point%20is%20deprecated.-,FAQs,-React%20Native%20DevTools).

**Links**

- [Updated debugging docs](/docs/debugging)
- [Announcement talk at React Universe Conf](https://www.youtube.com/watch?v=OwivVpg6Luc)
- [Feedback thread and FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819)

### Box Shadow and Filter style props

> We’ve added two New Architecture only style props in 0.76 - `boxShadow` and `filter`. Both of these props exist on the web, and the team tried to stick to the specs when possible so that these props are predictable, familiar, and ultimately easier to adopt. As a result, you can look up the web documentation to fully understand how these work, but there are a few important differences which are noted below.

Reviewer 5: "and the team tried to stick to the specs when possible"

This sentence raises spidey senses. What does this mean? "We tried, but didn't stick to web specs"? "For everything we support, it matches the web specs"?

Can we make this more explicit?

-   SoLoader.init(this, false)
  }
}
```

This change is necessary in order to properly load `libreactnative.so` and is included in the upgrade-helper.

Library authors won’t be affected by this change, unless you have custom C++ code.
For a technical deep dive into this change, and suggestions for library authors, you can read more about it [in this dedicated post](https://github.com/react-native-community/discussions-and-proposals/discussions/816).

### Updates to Minimum iOS and Android SDK requirements

We have updated our minimum platform and SDK versions:

- iOS: [15.1](https://support.apple.com/en-gb/108051#151)
- Android: [SDK 24](https://developer.android.com/tools/releases/platforms#7.0) (Android 7)

This change was announced earlier in the year when 0.75 [was released](/blog/2024/08/12/release-0.75#last-version-supporting-minsdk-23-and-miniosversion-134). For more background, please see the dedicated posts for [Android](https://github.com/react-native-community/discussions-and-proposals/discussions/802) and [iOS](https://github.com/react-native-community/discussions-and-proposals/discussions/812).

### Other Breaking Changes

> - **Animation**

Reviewer 5: Why is this capitalized and the other bullets are lower case? Should they all be capitalized?

Reviewer 4: We capitalized proper modules (`Animation`, `Pressable`, ...) and we left generic concepts lowercase ("infra", "runtime").
We can change it, if we prefer for it to be all capital case.

  - Stop sending state updates to React in looping animation ([commit](https://github.com/facebook/react-native/commit/8af5e893c89b24dca972afe36ef442fd509d5804)). This was causing unnecessary re-rendering with looping animations.
- **devtools:**

>   - Remove Inspector Panel perf + network tabs under New Arch ([commit](https://github.com/facebook/react-native/commit/f220bde4d7d556b3be83cccff0eec5da22921aba)). ([RFC](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0777-remove-legacy-element-inspector-features.md))
> - **text engine**
>   - Always use AttributedStringBox instead of AttributedString in TextLayoutManager ([commit](https://github.com/facebook/react-native/commit/ee597bfe2bf7f11ac60dab87f47dbcb6e840749c))

Reviewer 2: nits:

- Formatting consistency. If there's any chance of adding a second sentence, prefer periods on every line.
- Also, looks like we changed the format of this section in the 0.75 post — are we including "(commit)" any more? 

```suggestion
  - Remove Inspector Panel perf + network tabs under New Arch ([commit](https://github.com/facebook/react-native/commit/f220bde4d7d556b3be83cccff0eec5da22921aba)) ([RFC](https://github.com/react-native-community/discussions-and-proposals/blob/main/proposals/0777-remove-legacy-element-inspector-features.md)).
- **Text engine**
  - Always use AttributedStringBox instead of AttributedString in TextLayoutManager ([commit](https://github.com/facebook/react-native/commit/ee597bfe2bf7f11ac60dab87f47dbcb6e840749c)).
```

Reviewer 4: We are using capital letters for proper modules (Animation, Pressable, Image) and lower case for the rest (infra, text engine, runtime).

#### iOS

- **turbomodule**
  - Removed the `RCT_EXPORT_CXX_MODULE_EXPERIMENTAL` Macro for autolinking pure Cxx module ([commit](https://github.com/facebook/react-native/commit/184646e49169110842bafa6c02e52f6eec1e0ffa#diff-28b37f7a1a36054d37cd957bd4718ce481006efb286e5e88dec565087c4a8727)).

## Acknowledgements

React Native 0.76 contains over [1052 commits](https://github.com/facebook/react-native/compare/6f7eae5c583090bd62ca60a4c3170445e5a0a751..0.76-stable) from 156 contributors. Thanks for all of your hard work!

Thanks to all the additional authors that worked on documenting features in this release post:

- [Joe Vilches](https://github.com/joevilches) and [Nick Gerleman](https://github.com/NickGerleman) box-shadow and filter style props.
- [Alex Hunt](https://github.com/huntie) React Native DevTools.
- [Nicola Corti](https://github.com/cortinico) Android shipping as a single library.

## Upgrade to 0.76

Please use the [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/) to view code changes between React Native versions for existing projects, in addition to the Upgrading docs.

> To create a new project:

Reviewer 5: probably need to be clear that this isn't the official way to create a project, it's the community cli. 

And since we recommend people create new projects with Expo, I'd pull up the Expo line to be first

Reviewer 4: The problem is that Expo is not ready yet to support 0.76. We need to wait for SDK 52. We can move up the expo sentence.

Phasing out logs in Metro
In our next release, we’re also removing forwarded logs in Metro. We are doing this to align with modern browser tooling and to remove old, non-CDP, debugging integrations. Streamed application logs are not a pattern in web bundlers or in native build tools, and these are neither well-formatted nor searchable. Instead, please use React Native DevTools and its fully featured Console panel. See more under our [FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819#:~:text=point%20is%20deprecated.-,FAQs,-React%20Native%20DevTools).

**Links**

- [Updated debugging docs](https://reactnative.dev/docs/debugging)
- [Announcement talk at React Universe Conf](https://www.youtube.com/watch?v=OwivVpg6Luc)
- [Feedback thread and FAQs](https://github.com/react-native-community/discussions-and-proposals/discussions/819)

### Box Shadow and Filter style props

> We’ve added two New-Architecture-only style props in 0.76 - boxShadow and filter. Both of these props exist on the web, and the team tried to stick to the specs when possible so that these props are predictable, familiar, and ultimately easier to adopt. As a result you can look up the web documentation to fully understand how these work, but there are a few important differences which are noted below.

Reviewer 7: @blakef Can you stylize boxShadow and filter here so it's consistent with the rest of the document
